include/aspect/simulator_signals.h

```diff
@@ -287,6 +288,15 @@ namespace aspect
     boost::signals2::signal<void (const SimulatorAccess<dim> &,
                                   aspect::Assemblers::Manager<dim> &)>
     set_assemblers;
+
+    /**
+    * A signal that is called before the build_patches() function is called during
+    * the creation of the visualization output. This signal
+    * allows for registering functions that take a DataOut object and can for example
+    * be used to select only certain cells of the mesh to be built into patches through
+    * calling the DataOut member function set_cell_selection().
+    */
+    boost::signals2::signal<void (DataOut<dim> &)>  pre_data_out_build_patches;
```

Comment:
This is one way. I assume you want to use this to let the called function attach itself to the DataOut? Why not have a signal that returns a function object selecting individual cells that can then directly be attached to DoFHandler?

Reply:
As just discussed, we tried to keep the signal as general as possible as other things could be done with DataOut.